tools/mason/MasonUpdate.chpl

```diff
@@ -84,7 +84,14 @@ proc updateLock(skipUpdate: bool, tf="Mason.toml", lf="Mason.lock") {
     const projectHome = getProjectHome(cwd, tf);
     const tomlPath = projectHome + "/" + tf;
     const lockPath = projectHome + "/" + lf;
-    updateRegistry(skipUpdate);
+    
+    if isFile(lockPath) {
```

Comment:
If I understand correctly, this will require an update the first time a user runs `mason build` or `mason update`, then will skip further updates if the lock file exists and has no dependencies.

That seems good, although I'm wondering why not check the manifest file instead? That way, we never have to update the registries as long as there are no dependencies.

Reply:
Actually, we already have a `[dependencies]` table header at the manifest, and I couldn't find a method to check if that header has any contents.

Reply:
That's a good point. I have created an issue to document this shortcoming: https://github.com/chapel-lang/chapel/issues/16189

As a work-around, you should be able to inspect `Toml.A.size` (where `A` is the map of keys to values within the table).

Reply:
> As a work-around, you should be able to inspect Toml.A.size (where A is the map of keys to values within the table).

@ben-albrecht 
```chapel
const openToml = openreader(tomlPath);
const tomlFile = parseToml(openToml);
writeln(tomlFile.A)
```
returns 
```
{brick: name = "asd"
version = "0.1.0"
chplVersion = "1.23.0"
license = "None"

, dependencies:
}
```
here, the value of `tomlFile.A.size` is `2`(maybe because of 2 headers). 
But I am still unable to access the number of elements within a header. for example, 
`tomlFile.A['brick']` gives output : 
```
name = "project"
version = "0.1.0"
chplVersion = "1.23.0"
license = "None"
```
But I get an error on `tomlFile.A['brick'].size`  as follows: 
```
./MasonUpdate.chpl:80: In function 'updateLock':
./MasonUpdate.chpl:98: error: unresolved call 'unmanaged Toml?.size'
$CHPL_HOME/modules/internal/CString.chpl:187: note: this candidate did not match: c_string.sizereturn
./MasonUpdate.chpl:98: note: because method call receiver with type unmanaged Toml?
$CHPL_HOME/modules/internal/CString.chpl:187: note: is passed to formal 'this: c_string'
$CHPL_HOME/modules/internal/String.chpl:1082: note: candidates are: const string.sizereturn
$CHPL_HOME/modules/internal/String.chpl:222: note:                 __serializeHelper.size
note: and 38 other candidates, use --print-all-candidates to see them
```

Reply:
Can you try `tomlFile.A['brick'].A.size` ?

Reply:
I still get an error there.

Reply:
This should work:

```chpl
writeln(tomlFile['brick']!.A.size);
```